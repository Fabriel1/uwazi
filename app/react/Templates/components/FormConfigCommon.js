import React, {Component, PropTypes} from 'react';
import {Field} from 'react-redux-form';
import {connect} from 'react-redux';

import PrioritySortingLabel from './PrioritySortingLabel';

export class FormConfigCommon extends Component {

  render() {
    const {index} = this.props;
    const baseZeroIndex = index + this.props.data.commonProperties.length;

    return (
      <Field model={`template.data.commonProperties[${baseZeroIndex}].prioritySorting`}>
        <input id={'prioritySorting' + this.props.index} type="checkbox" />
        &nbsp;
        <PrioritySortingLabel htmlFor={'prioritySorting' + this.props.index} />
      </Field>
    );
  }
}

FormConfigCommon.propTypes = {
  data: PropTypes.object,
  index: PropTypes.number
};

export function mapStateToProps({template}) {
  return {
    data: template.data
  };
}

export default connect(mapStateToProps)(FormConfigCommon);
