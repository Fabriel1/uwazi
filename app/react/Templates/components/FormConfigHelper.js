import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { Field } from 'react-redux-form';
import { connect } from 'react-redux';

class FormConfigHelper extends Component {
  render() {
    const { index, formState, canShowInCard, helpText } = this.props;
    let labelClass = 'form-group';
    const labelKey = `properties.${index}.label`;
    const requiredLabel = formState.$form.errors[`${labelKey}.required`];
    const duplicatedLabel = formState.$form.errors[`${labelKey}.duplicated`];
    if (requiredLabel || duplicatedLabel) {
      labelClass += ' has-error';
    }

    return (
      <div>
        <div className={labelClass}>
          <label>Name</label>
          <Field model={`template.data.properties[${index}].label`}>
            <input className="form-control"/>
          </Field>
        </div>

        {helpText &&
          <div className="protip">
            <i className="fa fa-lightbulb-o"/>
            <span>This is an automatically generated field.  For Documents, it renders a thumbnail of the PDFs first page.</span>
          </div>
        }

        {canShowInCard &&
          <Field model={`template.data.properties[${index}].showInCard`}>
            <input id={`showInCard${this.props.index}`} type="checkbox"/>
            &nbsp;
            <label className="property-label" htmlFor={`showInCard${this.props.index}`}>
              Show in cards
              <i className="property-help fa fa-question-circle">
                <div className="property-description">Show this property in the cards as part of the basic info.</div>
              </i>
            </label>
          </Field>
        }
      </div>
    );
  }
}

FormConfigHelper.defaultProps = {
  canShowInCard: true,
  helpText: ''
};

FormConfigHelper.propTypes = {
  canShowInCard: PropTypes.bool,
  helpText: PropTypes.string,
  index: PropTypes.number.isRequired,
  formState: PropTypes.instanceOf(Object).isRequired
};

export function mapStateToProps({ template }) {
  return {
    formState: template.formState
  };
}

export default connect(mapStateToProps)(FormConfigHelper);
