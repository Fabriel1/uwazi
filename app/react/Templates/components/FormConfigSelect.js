import React, {Component, PropTypes} from 'react';
import FilterSuggestions from 'app/Templates/components/FilterSuggestions';
import {Select} from 'app/Forms';
import {connect} from 'react-redux';
import {Field} from 'react-redux-form';

export class FormConfigSelect extends Component {

  contentValidation() {
    return {required: (val) => val.trim() !== ''};
  }

  render() {
    const {index, data, formState} = this.props;
    const thesauris = this.props.thesauris.toJS();
    const ptoperty = data.properties[index];

    let optionGroups = [
      {label: 'Dictionaries', options: []},
      {label: 'Entities', options: []}
    ];

    thesauris.filter((thesauri) => {
      return thesauri._id !== data._id;
    }).forEach((thesauri) => {
      if (thesauri.type === 'template') {
        optionGroups[1].options.push(thesauri);
        return;
      }
      optionGroups[0].options.push(thesauri);
    });

    let labelClass = 'input-group';
    let labelKey = `properties.${index}.label`;
    let requiredLabel = formState.$form.errors[labelKey + '.required'];
    let duplicatedLabel = formState.$form.errors[labelKey + '.duplicated'];
    if (requiredLabel || duplicatedLabel) {
      labelClass += ' has-error';
    }

    return (
      <div>
        <div className="row">
          <div className="col-sm-12">
            <div className={labelClass}>
              <span className="input-group-addon">
              Label
              </span>
              <Field model={`template.data.properties[${index}].label`}>
                <input className="form-control"/>
              </Field>
            </div>
          </div>
        </div>
        <div className="row">
          <div className="col-sm-12">
            <div className="input-group">
              <span className="input-group-addon">Thesauri</span>
              <Select model={`template.data.properties[${index}].content`} options={optionGroups} optionsLabel="name" optionsValue="_id"/>
            </div>
          </div>
        </div>
        <div className="row">
          <div className="col-sm-4">
            <div className="input-group">
              <span className="input-group-addon">
                <Field model={`template.data.properties[${index}].required`}>
                  <input id={'required' + this.props.index} type="checkbox"/>
                </Field>
              </span>
              <label htmlFor={'required' + this.props.index} className="form-control">Required</label>
            </div>
          </div>
          <div className="col-sm-4">
            <div className="input-group">
              <span className="input-group-addon">
                <Field model={`template.data.properties[${index}].showInCard`}>
                  <input id={'showInCard' + this.props.index} type="checkbox"/>
                </Field>
              </span>
              <label htmlFor={'showInCard' + this.props.index}
                     className="form-control"
                     title="This property will appear in the library cards as part of the basic info.">
                Show in cards
                &nbsp;
                <i className="fa fa-question-circle"></i>
              </label>
            </div>
          </div>
        </div>

        <div className="well-metadata-creator">
          <div>
            <div>
              <Field model={`template.data.properties[${index}].filter`}>
                <input id={'filter' + this.props.index} type="checkbox"/>
              </Field>
              &nbsp;
              <label htmlFor={'filter' + this.props.index} title="This property will be used for filtering the library results.
              When properties match in equal name and field type with other document types, they will be combined for filtering.">
                Use as filter
                &nbsp;
                <i className="fa fa-question-circle"></i>
              </label>
            </div>
            <div>
              <FilterSuggestions {...ptoperty} />
            </div>
          </div>
        </div>

      </div>
    );
  }
}

FormConfigSelect.propTypes = {
  thesauris: PropTypes.object,
  data: PropTypes.object,
  index: PropTypes.number,
  formState: PropTypes.object,
  formKey: PropTypes.string
};

export function mapStateToProps(state) {
  return {
    data: state.template.data,
    thesauris: state.thesauris,
    formState: state.template.formState
  };
}

export default connect(mapStateToProps)(FormConfigSelect);
