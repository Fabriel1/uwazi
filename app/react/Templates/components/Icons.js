export default {
  checkbox: 'fa fa-check-square-o fa-fw',
  numeric: 'fa fa-calculator fa-fw',
  relationship: 'fa fa-link fa-fw',
  select: 'fa fa-sort fa-fw',
  multiselect: 'fa fa-list fa-fw',
  date: 'far fa-calendar fa-fw',
  text: 'fa fa-font fa-fw',
  markdown: 'far fa-file-alt fa-fw',
  nested: 'fa fa-cubes fa-fw',
  multidate: 'far fa-calendar fa-fw',
  multidaterange: 'far fa-calendar fa-fw',
  daterange: 'far fa-calendar fa-fw',
  geolocation: 'fa fa-map-marker fa-fw'
};
