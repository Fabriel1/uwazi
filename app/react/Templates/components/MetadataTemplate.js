import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { DropTarget } from 'react-dnd';
import { Form, Field, actions as formActions } from 'react-redux-form';

import { I18NLink } from 'app/I18N';
import { FormGroup } from 'app/Forms';
import ShowIf from 'app/App/ShowIf';

import { inserted, addProperty } from 'app/Templates/actions/templateActions';
import MetadataProperty from 'app/Templates/components/MetadataProperty';
import RemovePropertyConfirm from 'app/Templates/components/RemovePropertyConfirm';
import validator from './ValidateTemplate';

export class MetadataTemplateComponent extends Component {
  render() {
    const { connectDropTarget } = this.props;
    const commonProperties = this.props.commonProperties || [];

    return (
      <div>
        <RemovePropertyConfirm />
        <Form
          model="template.data"
          onSubmit={this.props.saveTemplate}
          className="metadataTemplate"
          validators={validator(this.props.properties, this.props.templates.toJS(), this.props._id)}
        >
          <div className="metadataTemplate-heading">
            <FormGroup model=".name">
              <Field model=".name">
                <input placeholder="Template name" className="form-control"/>
              </Field>
            </FormGroup>
          </div>

          <ShowIf if={!this.props.relationType}>
            {connectDropTarget(
              <ul className="metadataTemplate-list list-group">
                {commonProperties.map((config, index) => {
                  const localID = config.localID || config._id;
                  return <MetadataProperty {...config} key={localID} localID={localID} index={index - this.props.commonProperties.length} />;
                })}
                {this.props.properties.map((config, index) => {
                  const localID = config.localID || config._id;
                  return <MetadataProperty {...config} key={localID} localID={localID} index={index}/>;
                })}
                <div className="no-properties">
                  <span className="no-properties-wrap"><i className="fa fa-clone" />Drag properties here</span>
                </div>
              </ul>
            )}
          </ShowIf>
          <div className="settings-footer">
            <I18NLink to={this.props.backUrl} className="btn btn-default">
              <i className="fa fa-arrow-left" />
              <span className="btn-label">Back</span>
            </I18NLink>
            <button type="submit" className="btn btn-success save-template" disabled={!!this.props.savingTemplate}>
              <i className="fa fa-save"/>
              <span className="btn-label">Save</span>
            </button>
          </div>
        </Form>
      </div>
    );
  }
}

MetadataTemplateComponent.propTypes = {
  connectDropTarget: PropTypes.func.isRequired,
  formState: PropTypes.object,
  backUrl: PropTypes.string,
  _id: PropTypes.string,
  saveTemplate: PropTypes.func,
  savingTemplate: PropTypes.bool,
  relationType: PropTypes.bool,
  setErrors: PropTypes.func,
  properties: PropTypes.array,
  commonProperties: PropTypes.array,
  templates: PropTypes.object
};

const target = {
  canDrop() {
    return true;
  },

  drop(props, monitor) {
    const item = monitor.getItem();

    const propertyAlreadyAdded = props.properties[item.index];

    if (propertyAlreadyAdded) {
      props.inserted(item.index);
      return;
    }

    props.addProperty({ label: item.label, type: item.type }, props.properties.length);
    return { name: 'container' };
  }
};

const dropTarget = DropTarget('METADATA_OPTION', target, connector => ({
  connectDropTarget: connector.dropTarget()
}))(MetadataTemplateComponent);

export { dropTarget };

const mapStateToProps = ({ template, templates }) => ({
    _id: template.data._id,
    commonProperties: template.data.commonProperties,
    properties: template.data.properties,
    templates,
    savingTemplate: template.uiState.get('savingTemplate')
});

function mapDispatchToProps(dispatch) {
  return bindActionCreators({ inserted, addProperty, setErrors: formActions.setErrors }, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps, null, { withRef: true })(dropTarget);
