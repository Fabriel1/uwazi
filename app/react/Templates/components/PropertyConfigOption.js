import { Field } from 'react-redux-form';
import PropTypes from 'prop-types';
import React from 'react';

const PropertyConfigOption = ({ children, model, label }) => (
  <Field model={model}>
    <label className="property-label" htmlFor={`test${model}`}>
      <input id={`test${model}`} type="checkbox"/> {label}
    </label>
    {children}
  </Field>
);

PropertyConfigOption.defaultProps = {
  children: ''
};

PropertyConfigOption.propTypes = {
  children: PropTypes.node,
  model: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired
};

export default PropertyConfigOption;
