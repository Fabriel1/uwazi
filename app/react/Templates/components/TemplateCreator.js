import PropTypes from 'prop-types';
import React, {Component} from 'react';
import {DragDropContext} from 'react-dnd';
import HTML5Backend from 'react-dnd-html5-backend';
import {bindActionCreators} from 'redux';
import {connect} from 'react-redux';

import {resetTemplate, saveTemplate, saveEntity} from 'app/Templates/actions/templateActions';
import PropertyOption from 'app/Templates/components/PropertyOption';
import MetadataTemplate from 'app/Templates/components/MetadataTemplate';
import 'app/Templates/scss/templates.scss';
import ShowIf from 'app/App/ShowIf';

export class TemplateCreator extends Component {

  componentWillUnmount() {
    this.props.resetTemplate();
  }

  render() {
    let save = this.props.saveTemplate;
    let backUrl = '/settings/documents';
    if (this.props.entity) {
      save = this.props.saveEntity;
      backUrl = '/settings/entities';
    }

    return (
      <div className="metadata">
        <div className="panel panel-default">
          <div className="panel-heading">
            Metadata creator
          </div>
          <div className="panel-body">
            <div className="row">
              <main className="col-xs-12 col-sm-9">
                <MetadataTemplate saveTemplate={save} backUrl={backUrl} />
              </main>
              <aside className="col-xs-12 col-sm-3">
                <div className="metadataTemplate-constructor">
                  <div><i>Properties</i></div>
                  <ul className="list-group">
                    <PropertyOption label='Text' type='text'/>
                    <PropertyOption label='Numeric' type='numeric'/>
                    <PropertyOption label='Select' type='select' disabled={this.props.noThesauris} />
                    <PropertyOption label='Multi Select' type='multiselect' disabled={this.props.noThesauris} />
                    <PropertyOption label='Date' type='date'/>
                    <PropertyOption label='Date Range' type='daterange'/>
                    <PropertyOption label='Multi Date' type='multidate'/>
                    <PropertyOption label='Multi Date Range' type='multidaterange'/>
                    <PropertyOption label='Rich Text' type='markdown'/>
                    <ShowIf if={this.props.settings.collection.toJS().project === 'cejil'}>
                      <PropertyOption label='Violated articles' type='nested'/>
                    </ShowIf>
                  </ul>
                  <ShowIf if={this.props.noThesauris}>
                    <div className="alert alert-warning">
                      Selects and Multiselects can not be added untill you have at least one thesauri to select.
                    </div>
                  </ShowIf>
                </div>
              </aside>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

TemplateCreator.propTypes = {
  resetTemplate: PropTypes.func,
  saveTemplate: PropTypes.func,
  saveEntity: PropTypes.func,
  entity: PropTypes.bool,
  noThesauris: PropTypes.bool,
  settings: PropTypes.object
};

TemplateCreator.contextTypes = {
  router: PropTypes.object
};

function mapDispatchToProps(dispatch) {
  return bindActionCreators({resetTemplate, saveTemplate, saveEntity}, dispatch);
}

const mapStateToProps = ({settings, thesauris}) => {
  return {
    settings,
    noThesauris: !thesauris.size
  };
};

export default DragDropContext(HTML5Backend)(
  connect(mapStateToProps, mapDispatchToProps)(TemplateCreator)
);
