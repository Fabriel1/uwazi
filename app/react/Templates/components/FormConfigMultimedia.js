import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { Field } from 'react-redux-form';
import { connect } from 'react-redux';

import { t } from 'app/I18N';
import { Select } from 'app/ReactReduxForms';
import Tip from 'app/Layout/Tip';

import PropertyConfigOption from './PropertyConfigOption';

const style = index => (
  <div>
    <div className="form-group">
      <label>
        {t('System', 'Style')}
      </label>
      <Select
        model={`template.data.properties[${index}].style`}
        options={[{ _id: 'contain', name: 'Contain' }, { _id: 'cover', name: 'Cover' }]}
        optionsLabel="name"
        optionsValue="_id"
      />
    </div>
    <div className="protip">
      <p>
        <b>Contain</b> will show the entire media inside the container.  Grey bars may appear to the sides of the image.
        <br /><b>Cover</b> will attempt to fill the container, using it&#39;s entire width.  Cropping is likely to occur.
      </p>
    </div>
  </div>
);

class FormConfigMultimedia extends Component {
  render() {
    const { property, index, formState, canShowInCard, helpText } = this.props;

    let labelClass = 'form-group';
    const labelKey = `properties.${index}.label`;
    const requiredLabel = formState.$form.errors[`${labelKey}.required`];
    const duplicatedLabel = formState.$form.errors[`${labelKey}.duplicated`];
    if (requiredLabel || duplicatedLabel) {
      labelClass += ' has-error';
    }

    return (
      <div>
        <div className={labelClass}>
          <label>Name</label>
          <Field model={`template.data.properties[${index}].label`}>
            <input className="form-control"/>
          </Field>
        </div>

        {helpText &&
          <div className="protip">
            <i className="fa fa-lightbulb-o"/>
            <span>{helpText}</span>
          </div>
        }

        <PropertyConfigOption label="Show label" model={`template.data.properties[${index}].showLabel`}>
          <Tip>If selected, the property will be rendered with the label. Otherwise, it will just show the multimedia content.</Tip>
        </PropertyConfigOption>

        {canShowInCard &&
          <div>
            <PropertyConfigOption label="Show in cards" model={`template.data.properties[${index}].showInCard`}>
              <Tip>This property will appear in the library cards as part of the basic info.</Tip>
            </PropertyConfigOption>
            {property.showInCard && style(index)}
          </div>
        }
      </div>
    );
  }
}

FormConfigMultimedia.defaultProps = {
  canShowInCard: true,
  helpText: ''
};

FormConfigMultimedia.propTypes = {
  property: PropTypes.instanceOf(Object).isRequired,
  canShowInCard: PropTypes.bool,
  helpText: PropTypes.string,
  index: PropTypes.number.isRequired,
  formState: PropTypes.instanceOf(Object).isRequired
};

export function mapStateToProps({ template }, ownProps) {
  return {
    property: template.data.properties[ownProps.index],
    formState: template.formState
  };
}

export default connect(mapStateToProps)(FormConfigMultimedia);
