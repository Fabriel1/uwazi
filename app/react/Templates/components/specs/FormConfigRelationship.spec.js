import Immutable from 'immutable';
import React from 'react';

import { FormConfigRelationship } from 'app/Templates/components/FormConfigRelationship';
import { shallow } from 'enzyme';

describe('FormConfigRelationship', () => {
  let component;
  let thesauris;
  let relationTypes;
  let props;

  beforeEach(() => {
    thesauris = [{ _id: 1, name: 'thesauri1' }, { _id: 2, name: 'thesauri2' }, { _id: 3, name: 'Judge', type: 'template' }];
    relationTypes = [{ _id: 1, name: 'relationType1' }, { _id: 2, name: 'relationType2' }];
    props = {
      thesauris: Immutable.fromJS(thesauris),
      relationTypes: Immutable.fromJS(relationTypes),
      index: 0,
      data: { properties: [{ filter: false }] },
      formState: {
        'properties.0.label': { valid: true, dirty: false, errors: {} },
        $form: {
          errors: {
            'properties.0.label.required': false,
            'properties.0.label.duplicated': false
          }
        }
      }
    };
  });

  it('should render fields with the correct datas', () => {
    component = shallow(<FormConfigRelationship {...props}/>);
    expect(component).toMatchSnapshot();
  });

  describe('when the fields are invalid and dirty or the form is submited', () => {
    it('should render the label with errors', () => {
      props.formState.$form.errors['properties.0.label.required'] = true;
      props.formState['properties.0.label'].touched = true;
      component = shallow(<FormConfigRelationship {...props}/>);
      expect(component).toMatchSnapshot();
    });

    it('should render the label with errors when the form is submited', () => {
      props.formState.$form.errors['properties.0.label.required'] = true;
      props.formState.submitFailed = true;
      component = shallow(<FormConfigRelationship {...props}/>);
      expect(component).toMatchSnapshot();
    });
  });

  describe('when use as filter is selected', () => {
    it('should show the default filter option', () => {
      props.data.properties[0].filter = true;
      component = shallow(<FormConfigRelationship {...props}/>);
      expect(component).toMatchSnapshot();
    });
  });
});
