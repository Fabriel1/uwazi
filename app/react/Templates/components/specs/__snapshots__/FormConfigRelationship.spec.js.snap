// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormConfigRelationship should render fields with the correct datas 1`] = `
<div>
  <div
    className="form-group"
  >
    <label>
      Label
    </label>
    <Modeled(Connect(Field))
      model="template.data.properties[0].label"
    >
      <input
        className="form-control"
      />
    </Modeled(Connect(Field))>
  </div>
  <div
    className="form-group"
  >
    <label>
      Select list
    </label>
    <Select
      model="template.data.properties[0].content"
      options={
        Array [
          Object {
            "_id": 3,
            "name": "Judge",
            "type": "template",
          },
        ]
      }
      optionsLabel="name"
      optionsValue="_id"
      placeholder="Any entity or document"
    />
  </div>
  <div
    className="form-group"
  >
    <label>
      Relationship
      <span
        className="required"
      >
        *
      </span>
    </label>
    <Select
      model="template.data.properties[0].relationType"
      options={
        Array [
          Object {
            "_id": 1,
            "name": "relationType1",
          },
          Object {
            "_id": 2,
            "name": "relationType2",
          },
        ]
      }
      optionsLabel="name"
      optionsValue="_id"
      validators={
        Object {
          "required": [Function],
        }
      }
    />
  </div>
  <PropertyConfigOption
    label="Required property"
    model="template.data.properties[0].required"
  >
    <Tip>
      You won't be able to publish a document if this property is empty.
    </Tip>
  </PropertyConfigOption>
  <PropertyConfigOption
    label="Show in cards"
    model="template.data.properties[0].showInCard"
  >
    <Tip>
      This property will appear in the library cards as part of the basic info.
    </Tip>
  </PropertyConfigOption>
  <div>
    <PropertyConfigOption
      label="Use as filter"
      model="template.data.properties[0].filter"
    >
      <Tip>
        This property will be used for filtering the library results. When properties match in equal name and field type with other document types, they will be combined for filtering.
      </Tip>
    </PropertyConfigOption>
    <Connect(FilterSuggestions)
      filter={false}
    />
  </div>
</div>
`;

exports[`FormConfigRelationship when the fields are invalid and dirty or the form is submited should render the label with errors 1`] = `
<div>
  <div
    className="form-group has-error"
  >
    <label>
      Label
    </label>
    <Modeled(Connect(Field))
      model="template.data.properties[0].label"
    >
      <input
        className="form-control"
      />
    </Modeled(Connect(Field))>
  </div>
  <div
    className="form-group"
  >
    <label>
      Select list
    </label>
    <Select
      model="template.data.properties[0].content"
      options={
        Array [
          Object {
            "_id": 3,
            "name": "Judge",
            "type": "template",
          },
        ]
      }
      optionsLabel="name"
      optionsValue="_id"
      placeholder="Any entity or document"
    />
  </div>
  <div
    className="form-group"
  >
    <label>
      Relationship
      <span
        className="required"
      >
        *
      </span>
    </label>
    <Select
      model="template.data.properties[0].relationType"
      options={
        Array [
          Object {
            "_id": 1,
            "name": "relationType1",
          },
          Object {
            "_id": 2,
            "name": "relationType2",
          },
        ]
      }
      optionsLabel="name"
      optionsValue="_id"
      validators={
        Object {
          "required": [Function],
        }
      }
    />
  </div>
  <PropertyConfigOption
    label="Required property"
    model="template.data.properties[0].required"
  >
    <Tip>
      You won't be able to publish a document if this property is empty.
    </Tip>
  </PropertyConfigOption>
  <PropertyConfigOption
    label="Show in cards"
    model="template.data.properties[0].showInCard"
  >
    <Tip>
      This property will appear in the library cards as part of the basic info.
    </Tip>
  </PropertyConfigOption>
  <div>
    <PropertyConfigOption
      label="Use as filter"
      model="template.data.properties[0].filter"
    >
      <Tip>
        This property will be used for filtering the library results. When properties match in equal name and field type with other document types, they will be combined for filtering.
      </Tip>
    </PropertyConfigOption>
    <Connect(FilterSuggestions)
      filter={false}
    />
  </div>
</div>
`;

exports[`FormConfigRelationship when the fields are invalid and dirty or the form is submited should render the label with errors when the form is submited 1`] = `
<div>
  <div
    className="form-group has-error"
  >
    <label>
      Label
    </label>
    <Modeled(Connect(Field))
      model="template.data.properties[0].label"
    >
      <input
        className="form-control"
      />
    </Modeled(Connect(Field))>
  </div>
  <div
    className="form-group"
  >
    <label>
      Select list
    </label>
    <Select
      model="template.data.properties[0].content"
      options={
        Array [
          Object {
            "_id": 3,
            "name": "Judge",
            "type": "template",
          },
        ]
      }
      optionsLabel="name"
      optionsValue="_id"
      placeholder="Any entity or document"
    />
  </div>
  <div
    className="form-group"
  >
    <label>
      Relationship
      <span
        className="required"
      >
        *
      </span>
    </label>
    <Select
      model="template.data.properties[0].relationType"
      options={
        Array [
          Object {
            "_id": 1,
            "name": "relationType1",
          },
          Object {
            "_id": 2,
            "name": "relationType2",
          },
        ]
      }
      optionsLabel="name"
      optionsValue="_id"
      validators={
        Object {
          "required": [Function],
        }
      }
    />
  </div>
  <PropertyConfigOption
    label="Required property"
    model="template.data.properties[0].required"
  >
    <Tip>
      You won't be able to publish a document if this property is empty.
    </Tip>
  </PropertyConfigOption>
  <PropertyConfigOption
    label="Show in cards"
    model="template.data.properties[0].showInCard"
  >
    <Tip>
      This property will appear in the library cards as part of the basic info.
    </Tip>
  </PropertyConfigOption>
  <div>
    <PropertyConfigOption
      label="Use as filter"
      model="template.data.properties[0].filter"
    >
      <Tip>
        This property will be used for filtering the library results. When properties match in equal name and field type with other document types, they will be combined for filtering.
      </Tip>
    </PropertyConfigOption>
    <Connect(FilterSuggestions)
      filter={false}
    />
  </div>
</div>
`;

exports[`FormConfigRelationship when use as filter is selected should show the default filter option 1`] = `
<div>
  <div
    className="form-group"
  >
    <label>
      Label
    </label>
    <Modeled(Connect(Field))
      model="template.data.properties[0].label"
    >
      <input
        className="form-control"
      />
    </Modeled(Connect(Field))>
  </div>
  <div
    className="form-group"
  >
    <label>
      Select list
    </label>
    <Select
      model="template.data.properties[0].content"
      options={
        Array [
          Object {
            "_id": 3,
            "name": "Judge",
            "type": "template",
          },
        ]
      }
      optionsLabel="name"
      optionsValue="_id"
      placeholder="Any entity or document"
    />
  </div>
  <div
    className="form-group"
  >
    <label>
      Relationship
      <span
        className="required"
      >
        *
      </span>
    </label>
    <Select
      model="template.data.properties[0].relationType"
      options={
        Array [
          Object {
            "_id": 1,
            "name": "relationType1",
          },
          Object {
            "_id": 2,
            "name": "relationType2",
          },
        ]
      }
      optionsLabel="name"
      optionsValue="_id"
      validators={
        Object {
          "required": [Function],
        }
      }
    />
  </div>
  <PropertyConfigOption
    label="Required property"
    model="template.data.properties[0].required"
  >
    <Tip>
      You won't be able to publish a document if this property is empty.
    </Tip>
  </PropertyConfigOption>
  <PropertyConfigOption
    label="Show in cards"
    model="template.data.properties[0].showInCard"
  >
    <Tip>
      This property will appear in the library cards as part of the basic info.
    </Tip>
  </PropertyConfigOption>
  <div>
    <PropertyConfigOption
      label="Use as filter"
      model="template.data.properties[0].filter"
    >
      <Tip>
        This property will be used for filtering the library results. When properties match in equal name and field type with other document types, they will be combined for filtering.
      </Tip>
    </PropertyConfigOption>
    <PropertyConfigOption
      label="Default filter"
      model="template.data.properties[0].defaultfilter"
    >
      <Tip>
        Use this property as a default filter in the library. When there are no document types selected, this property will show as a default filter for your collection.
      </Tip>
    </PropertyConfigOption>
    <Connect(FilterSuggestions)
      filter={true}
    />
  </div>
</div>
`;
